### ISSUE TITLE
Teleport Connect Blank Window

### ISSUE THREAD

**Issue by JasonEleventeen** [gravitational/teleport#32832]
Expected behavior: 
window displays data

Current behavior: 
window is blank

Bug details:
version 14.0.1
OS Ubuntu 23.04, Gnome Wayland

Window blanks out of a couple of seconds, user input appears to have no effect

I've tried deleting "~/.config/Teleport Connect" and reinstalling and older versions back to 13, same result

![image](https://github.com/gravitational/teleport/assets/19441652/383cb8bf-c227-46a6-a996-823b78f8b7f0)

**Comment by JasonEleventeen** [comment]
cli output 
`$ teleport-connect 

[121297:1002/104840.680352:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[121297:1002/104842.454567:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!`

**Comment by ravicious** [comment]
Could you include logs? https://goteleport.com/docs/connect-your-client/teleport-connect/#submitting-an-issue

**Comment by JasonEleventeen** [comment]
[logs.zip](https://github.com/gravitational/teleport/files/12789732/logs.zip)

**Comment by pandreani-opendsi** [comment]
Same problem.

Linux Mint 21.2
Cinnamon X11
Teleport version 13.4.1

No information in the logs.
Remove the .config/Teleport Connect

Nothing change, blank window.

**Comment by ravicious** [comment]
Thanks for the logs! It appears that the frontend app does get executed (hence the logged requests in renderer.log), but the UI does not get rendered for some reason.

@pandreani-opendsi Do you also see similar messages in renderer.log? What is the output if you run `teleport-connect` from the terminal?

---

I couldn't reproduce this on Xubuntu 23.04. I'm trying to establish if it might be related to an Electron update in 13.4.0 and 14.0.0. Could you try running Connect 13.3.8 and tell me if the UI gets rendered there?

* https://cdn.teleport.dev/teleport-connect_13.3.8_amd64.deb
* https://cdn.teleport.dev/teleport-connect-13.3.8-x64.tar.gz

**Comment by ravicious** [comment]
Could you try to start the app from the terminal with `ELECTRON_ENABLE_STACK_DUMPING=1`?

```
ELECTRON_ENABLE_STACK_DUMPING=1 teleport-connect
```

If it does report a crash (like here: https://github.com/electron/electron/issues/39775#issuecomment-1732114473), then you're affected by that Electron issue.

If it doesn't report any crashes, then it might be that there's a bug in our UI initialization logic. We suspect the app hangs at the step where it usually shows a modal asking about telemetry. Could you check if closing the app, replacing the contents of `~/.config/Teleport\ Connect/app_config.json` with the snippet below and launching the app again resolves the problem?

```json
{
  "$schema": "schema_app_config.json",
  "usageReporting.enabled": false
}
```

**Comment by JasonEleventeen** [comment]
```
ELECTRON_ENABLE_STACK_DUMPING=1 teleport-connect
[478600:1004/084246.394285:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
```
Added nothing to the logs

Changing "usageReporting.enabled": false did not fix the issue

[logs.zip](https://github.com/gravitational/teleport/files/12796419/logs.zip)

**Comment by pandreani-opendsi** [comment]
Nothing change.

**Comment by ravicious** [comment]
Could you also try running Connect 13.3.8? https://github.com/gravitational/teleport/issues/32832#issuecomment-1744829844 Identifying the last version that worked will help us narrow down the issue as well.

**Comment by ravicious** [comment]
We were able to reproduce the issue and we believe that it's due to [an issue in Electron v26](https://github.com/electron/electron/issues/39775). One of our teammates confirmed that downgrading Electron to v25 helps.

Until #33011 gets backported and released, you can use either 13.3.8 or the custom build of v14 that I made. [It's build off v14](https://github.com/gravitational/teleport/commits/v14.0.2-dev.ravicious.3) but with Electron downgraded to v25. These artifacts should show up in our CDN within 15-30 minutes of me posting this comment:

* https://cdn.teleport.dev/teleport-connect_14.0.2-dev.ravicious.3_amd64.deb
* https://cdn.teleport.dev/teleport-connect-14.0.2-dev.ravicious.3.x86_64.rpm
* https://cdn.teleport.dev/teleport-connect-14.0.2-dev.ravicious.3-x64.tar.gz

**Comment by gzdunek** [comment]
The fix will be available in the next patch versions for v14, v13 and v12.
